scripts/watch_dog.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/python
+"""This script will watch for an idle mic data and restart mycroft.
+
+This script will watch for an in active mic and restart pulseaudio and
+mycroft to recover for lost or reconnedted mic.  This takes a bit of time
+to recover so give it a couple of minutes.  This completely restarts
+mycroft.
+
+This requires setup of the mycroft user for ssh access as the pi user.
+Commands to restart mycroft are given as sudo's as the pi user with ssh:
+
+Example:
+    To start mycroft message bus
+    literal blocks::
+
+        $ ssh pi@localhost ./restart_mycroft.sh
+
+So mycroft user will need a ssh key generated and copied to authorized_keys.
+This will allow the mycroft user to ssh without password.  The following
+commands can be used set this up.
+    literal blocks::
+
+        $ ssh-keygen
+        $ ssh-copy-id pi@localhost
+
+Note these commands should be ran as the mycroft user.  The keygen command
+can be used just with defaults.  If you include a passcode then it will
+be required and cause the script not to work.  The second command the copy
+will copy your public key into the authorized_keys. This will require the
+password for pi before copying.  You then follow the instruction to from
+the command to check if it worked.  You should be fine.
+
+The process is as follows
+    1. Check For Changes in mic_level file
+    2. Once the changes stop for 20 sec
+        A. Kill and Restart pulseaudio
+        B. Kill ans restart mycroft
+        C. Check the mic file is changing up to 1 min.
+    3. Return to step one.
+
+"""
+import os.path
+import subprocess
+import time
+
+forever = True
+
+filename = "/tmp/mycroft/ipc/mic_level"
```

Comment:
I was debating on how much I wanted to tie it into the mycroft code base.  But this can be done,